paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/link/LinkConfirmationActivityTest.kt

```diff
@@ -0,0 +1,248 @@
+    private val application = ApplicationProvider.getApplicationContext<Application>()
+
+    @get:Rule
+    val intentsRule = IntentsRule()
+
+    @get:Rule
+    val testActivityRule = createTestActivityRule<PaymentElementConfirmationTestActivity>()
+
+    @get:Rule
+    val paymentConfigurationTestRule = PaymentConfigurationTestRule(
+        context = application,
+    )
+
+    @get:Rule
+    val featureFlagTestRule = FeatureFlagTestRule(
+        featureFlag = FeatureFlags.nativeLinkEnabled,
+        isEnabled = false
+    )
+
+    @get:Rule
+    val shampooRule = ShampooRule(200)
+
+    @Test
+    fun linkSucceeds() = test {
+        val paymentMethod = PaymentMethodFactory.card()
+
+        confirmationHandler.state.test {
+            awaitItem().assertIdle()
+
+            confirmationHandler.start(
+                ConfirmationHandler.Args(
+                    intent = PAYMENT_INTENT,
+                    confirmationOption = LINK_CONFIRMATION_OPTION,
+                )
+            )
+
+            intendingLinkToBeLaunched(
+                LINK_COMPLETE_CODE,
+            ) {
+                val convertedPaymentMethod = String(
+                    Base64.encode(
+                        PaymentMethodFactory
+                            .convertCardToJson(paymentMethod)
+                            .toString(2)
+                            .encodeToByteArray(),
+                        0
+                    ),
+                    Charsets.UTF_8,
+                )
+
+                setData(
+                    Uri.parse("https://link.com/redirect?link_status=complete&pm=$convertedPaymentMethod")
+                )
+            }
+
+            intendingPaymentConfirmationToBeLaunched(
+                InternalPaymentResult.Completed(PAYMENT_INTENT.copy(paymentMethod = paymentMethod))
+            )
+
+            val confirmingWithLink = awaitItem().assertConfirming()
+
+            assertThat(confirmingWithLink.option).isEqualTo(LINK_CONFIRMATION_OPTION)
+
+            intendedLinkToBeLaunched()
+
+            val confirmingWithSavedPaymentMethod = awaitItem().assertConfirming()
+
+            assertThat(confirmingWithSavedPaymentMethod.option)
+                .isEqualTo(
+                    PaymentMethodConfirmationOption.Saved(
+                        initializationMode = LINK_CONFIRMATION_OPTION.initializationMode,
+                        shippingDetails = LINK_CONFIRMATION_OPTION.shippingDetails,
+                        paymentMethod = paymentMethod,
+                        optionsParams = null,
+                    )
+                )
+
+            intendedPaymentConfirmationToBeLaunched()
+
+            val successResult = awaitItem().assertComplete().result.assertSucceeded()
+
+            assertThat(successResult.intent).isEqualTo(PAYMENT_INTENT.copy(paymentMethod = paymentMethod))
+            assertThat(successResult.deferredIntentConfirmationType).isNull()
+        }
+    }
```

Comment:
Another big win! Confirmation activity tests for `Link` is now possible. Can build on these in future PRs.